File: backend/src/main/java/Models/Post.java
package Models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String content;

    private String image;

    @ManyToOne
    private User author;

    private Date date;

    @Column(columnDefinition = "TEXT")
    @ElementCollection
    private List<String> tags;

    private boolean visibility;

}

File: application/lib/Models/Post.dart
import 'package:enicar_social_media/Models/User.dart';

class Post {
  int id;
  String content;
  String? image;
  User? author;
  DateTime date;
  List<String> tags;
  bool visibility;

  Post({
    required this.id,
    required this.content,
    required this.image,
    required this.author,
    required this.date,
    required this.tags,
    required this.visibility,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      content: json['content'],
      image: json['image'],
      author: User.fromJson(json['author']),
      date: DateTime.parse(json['date']),
      tags: List<String>.from(json['tags']),
      visibility: json['visibility'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'content': content,
      'image': image,
      'author': author!.toJson(),
      'date': date.toIso8601String(),
      'tags': tags,
      'visibility': visibility,
    };
  }
}
